gluten-core/src/main/scala/org/apache/gluten/execution/JoinExecTransformer.scala

```diff
@@ -167,7 +167,8 @@ trait HashJoinLikeExecTransformer extends BaseJoinExec with TransformSupport {
     }
   }
 
-  protected lazy val substraitJoinType: JoinRel.JoinType = SubstraitUtil.toSubstrait(joinType)
+  protected lazy val substraitJoinType: JoinRel.JoinType =
+    SubstraitUtil.toSubstrait(joinType, needSwitchChildren)
```

Comment:
Is it possible to make the variable abstract? Then we can move the change to CH backend.

Thanks.

Reply:
Velox has its implement

Reply:
I'll left this unchanged